app/src/main/java/ru/vssemikoz/newsfeed/newsfeed/NewsFeedFragment.java

```diff
@@ -0,0 +1,195 @@
+        categoryButton = root.findViewById(R.id.ib_category);
+        recyclerView = root.findViewById(R.id.rv_news_feed);
+        initRecyclerView();
+    }
+
+    @Override
+    public void onSaveInstanceState(@NonNull Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putSerializable(CURRENT_CATEGORY, presenter.getCategory());
+        outState.putBoolean(CURRENT_SHOW_FAVORITE, presenter.getShowFavorite());
+    }
+
+    @Override
+    public void onViewStateRestored(@Nullable Bundle savedInstanceState) {
+        super.onViewStateRestored(savedInstanceState);
+
+        if (savedInstanceState != null) {
+            Category category = (Category) savedInstanceState.getSerializable(CURRENT_CATEGORY);
+            Boolean showFavoriteNews = savedInstanceState.getBoolean(CURRENT_SHOW_FAVORITE);
+            presenter.setShowFavorite(showFavoriteNews);
+            presenter.setCategory(category);
+        }
+
+    }
+
+    private void initRecyclerView() {
+        adapter = new NewsFeedAdapter(context);
+        recyclerView.setLayoutManager(new LinearLayoutManager(context));
+        recyclerView.setAdapter(adapter);
+        adapter.setOnItemClickListener(new NewsFeedAdapter.OnNewsItemClickListener() {
+            @Override
+            public void onChangeFavoriteStateClick(int position) {
+                presenter.changeNewsFavoriteState(position);
+            }
+
+            @Override
+            public void OnRecyclerItemClick(int position) {
+                presenter.openNewsDetails(position, context);
+            }
+        });
+    }
+
+    @Override
+    public void setFavoriteIcon(Boolean showOnlyFavorite) {
+        if (showOnlyFavorite) {
+            favoriteNewsButton.setImageDrawable(IconicStorage.getYellowStarBorderless(context));
+        } else {
+            favoriteNewsButton.setImageDrawable(IconicStorage.getWhiteStarBorderless(context));
+        }
+    }
+
+    @Override
+    public void setCategoryTitle(Category category) {
+        updateCategoryNameOnDescription(Category.getDisplayName(category));
+    }
+
+    @Override
+    public void updateNewsItem(int position) {
+        adapter.notifyItemChanged(position);
+    }
+
+    @Override
+    public void removeNewsItem(int position) {
+        adapter.notifyItemRemoved(position);
+    }
+
+    @Override
+    public void showProgressBar() {
+        progressBar.setVisibility(ProgressBar.VISIBLE);
+    }
+
+    @Override
+    public void hideProgressBar() {
+        progressBar.setVisibility(ProgressBar.GONE);
+    }
+
+    @Override
+    public void showCategoryDialog() {
+        PickCategoryDialog categoryDialog = new PickCategoryDialog();
+        categoryDialog.setListener(this);
+        categoryDialog.show(Objects.requireNonNull(getActivity()).getSupportFragmentManager(),
+                "categoryDialog");
+    }
+
+    @Override
+    public void setEmptyViewOnDisplay() {
```

Comment:
Let's make things simpler like `showEmptyView` / `showList`. You shouldn't show too many framework details in your method names.